errors.go

```diff
@@ -109,6 +124,40 @@ func sprintf(format string, args ...interface{}) string {
 	return fmt.Sprintf(format, args...)
 }
 
+// NewBatch creates a new BatchError.
+func NewBatch() BatchError {
+	return &batcherrtype{}
+}
+
+//Append appends the error
+func (b *batcherrtype) Append(err error) {
+	b.errors = append(b.errors, err)
+}
+
+//Errors returns the errors
+func (b *batcherrtype) Errors() []error {
+	return b.errors
+}
+
+//IsEmpty returns if there is any error
+func (b *batcherrtype) IsEmpty() bool {
+	return len(b.errors) == 0
+}
```

Comment:
@RafaelReia indeed thats the idea. I like the api you made but it makes one thing against the language practices, which check errors for nil. 

If our error is _not nil_ but instead is _empty_, the traditional check `if err != nil { // handle error }` would be confusing. So maybe we can take the hit ux wise for more correctness in language practices.